stdlib/asm/math/u64.masm

```diff
@@ -673,6 +675,31 @@ export.unchecked_shr
     cswap
 end
 
+# Performs left shift of one unsigned 64-bit integer preserving the overflow and
+# using the pow2 operation.
+# The input value to be shifted is assumed to be represented using 32 bit limbs.
+# The shift value is assumed to be in the range [0, 64).
+# Stack transition looks as follows:
+# [b, a_hi, a_lo, ...] -> [d_hi, d_lo, c_hi, c_lo, ...], where (d,c) = a << b
```

Comment:
Which one is the better to represent this stack transition? Can we describe it even better?
```suggestion
# [b, a_hi, a_lo, ...] -> [c_hi, c_mid_hi, c_mid_lo, c_lo, ...], where c = a << b
```

Reply:
I like the first approach better, but I'd also add that `d` contains bits which were shifted out of `a` or something like that.